### PR TITLE
Implement Alchemist Artificer: Alchemical Savant

### PR DESCRIPTION
**Pull request by Aeristoka**
https://www.dndbeyond.com/classes/artificer#Alchemist

Alchemical Savant
At 5th level, you develop masterful command of magical chemicals, enhancing the healing and damage you create through them. Whenever you cast a spell using your alchemist’s supplies as the spellcasting focus, you gain a bonus to one roll of the spell. That roll must restore hit points or be a damage roll that deals acid, fire, necrotic, or poison damage, and the bonus equals your Intelligence modifier (minimum of +1).

### CODE REVIEW

File: src/common/settings.js [hunk 1]
```diff
@@ -451,6 +451,12 @@ const character_settings = {
         "description": "Unleash your divine soul to deal extra radiant damage equal to your level.",
         "type": "bool",
         "default": false
+    },
+    "artificer-alchemical-savant": {
+        "title": "Artificer: Use Alchemical Savant",
+        "description": "Use an Alchemical Savant for your Artificer spells. Deals extra damage equal to Intelligence Modifier",
```
Comment (kakaroto) [inline src/common/settings.js:457]: I'd rename the description as "Use your Alchemist's supplies as spellcasting focus, dealing extra damage or healing equal to your Intelligence Modifier."

File: src/dndbeyond/content-scripts/character.js [hunk 2]
```diff
@@ -631,6 +631,19 @@ function rollSpell(force_display = false) {
             damage_types.push("Arcane Firearm");
         }
 
+        if (character.hasClassFeature("Alchemical Savant") &&
+            character.getSetting("artificer-alchemical-savant", false) &&
+            damages.length) {
+            for (let i = 0; i < damages.length; i++){
+                if (damage_types[i] === "Acid" || damage_types[i] === "Fire" ||
+                    damage_types[i] === "Necrotic" || damage_types[i] === "Poison") {
+                    damages.push(`${character.getAbility("INT").mod < 2 ? 1 : character.getAbility("INT").mod}`);
```
Comment (kakaroto) [inline src/dndbeyond/content-scripts/character.js:640]: `Math.min(1, character.getAbility("INT").mod)` is more readable than the ternary operator here. Same for the healing part

**Review by kakaroto — COMMENTED** [review]
Couple of notes

File: src/dndbeyond/content-scripts/character.js (hunk 2 above)
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js:640]: Should be Math.max though: That roll must restore hit points or be a damage roll that deals acid, fire, necrotic, or poison damage, and the bonus equals your Intelligence modifier (minimum of +1).

https://www.dndbeyond.com/classes/artificer

Comment (kakaroto) [reply src/dndbeyond/content-scripts/character.js:640]: arggg, yeah, I told you I was tired.. I saw "minimum" and thought Math.min 🤦‍♂️

**Comment by Aeristoka** [comment]
Closed by 48dc8eb

**Pull request closed without merging**